### ISSUE TITLE
IDE complains about some range patterns starting with constants

### ISSUE THREAD

**Issue by grainednoise** [intellij-rust/intellij-rust#4762]
## Environment

* **IntelliJ Rust plugin version:**
0.2.112.2148-193 (problem was also present in the version before that one)

* **Rust toolchain version:**
RustUp 0.2.112.2148-193
Default host: x86_64-pc-windows-msvc
stable-x86_64-pc-windows-msvc (default)
rustc 1.39.0 (4560ea788 2019-11-04)

* **IDE name and version:**
IntelliJ IDEA 2019.3 (Community Edition)
Build #IC-193.5233.102, built on November 28, 2019

* **Operating system:**
Windows 7 6.1

## Problem description

Range patterns in a match arm of a match statement are flagged as parse errors in the editor when:

* The lower part of the range is a _constant_ (i.e. defined as such)

* Either an _exclusive_ range is used, or a _new-style inclusive_ range (e.g. X..11 or X..=10)

Other usages of ranges seem to work fine.

## Steps to reproduce

Put code below in editor. Where it says "Confuses the parser" the errors should
occur, starting at the first dot after the constant.

```rust

fn i_am_confused() {
    const X: i32 = 1;
    const Y: i32 = 20;

    let dummy1 = match 10 {
        X...11 => true,    //Works, but uses deprecated syntax
        X..=11 => false,   //Confuses the parser
        X..12 => false,    //Confuses the parser
        11..=Y => true,    //No problem
        11..Y => true,     //No problem
        _ => true
    };

    let dummy2 = X..11;  //No problem
    let dummy3 = X..=11; //No problem
}

```
